### PR TITLE
Fix: add timezone fallback to install script

### PR DESCRIPTION
**Pull request by Harald-Berghoff**
## Proposed change

The PR adresses two issues I came across when trying to run the installation script on a QNAP NAS, having installed QTS. Both issues can be overcome easy, but I propose to fix it.

1. the default timezone proposed during the installation script is empty. It's caused by the unavailable command timedatectl on QTS. I assume this is the case with many busybox based linux. When both ways to identify a default timezone are not available, it will default to GMT Europe/London.

2. After a fresh install via docker compose with an separated subnet the webserver fails to connect to a postgres db container. This is rooted in the host authentication method of postgres, being set to trust only for localhost (loopback interface). The change adds for all hosts the trust authentication method in the postgres container. 
I guess it could also be worth to change the db setup from the web container to use password authentication, and afterwards remove this option again from the compose file. But till then this should fix the setup issue.

Feel free to select only one change or edit it. You can also get back to me if you think something should be adapted.

If you are not ok with the changes, it's ok to reject the PR.

I have tested the changes, and they solve the issue with keeping everything else working. Of course I only tested installation.

(Closes) #4033 and #6318  (I guess addresses the issues mentioned there fits better)

## Type of change

- [X] Bug fix: non-breaking change which fixes an issue.
- [X] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [X] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [X] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [X] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docker/compose/docker-compose.postgres.yml
Comment (stumpylog) [inline docker/compose/docker-compose.postgres.yml:1]: We're not going to make a global change because of your specific setup.  This is also insecure:

> Note 1: It is not recommended to use trust since it allows anyone to connect without a password, even if one is set ([1](https://hub.docker.com/_/postgres))

Please revert this change.

File: install-paperless-ngx.sh [hunk 1]
```diff
@@ -71,7 +71,16 @@ if ! docker stats --no-stream &> /dev/null ; then
...
-default_time_zone=$(timedatectl show -p Timezone --value)
+# Added handling for timezone for busybox based linux, not having timedatectl available (i.e. QNAP QTS)
+# if neither timedatectl or /etc/TZ is suceeding, defaulting to GMT.
+if  command -v timedatectl &> /dev/null ; then
+        default_time_zone=$(timedatectl show -p Timezone --value)
+elif [ -f /etc/TZ ] && [ -f /etc/tzlist ] ; then
+        TZ=`cat /etc/TZ`
+        default_time_zone=$(grep -B 1 -m 1 $TZ /etc/tzlist | head -1 | cut -f 2 -d =)
+else
+        default_time_zone="Europe/London"
```
Comment (stumpylog) [inline install-paperless-ngx.sh:82]: - This should not use Europe/London, but [Etc/UTC](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#UTC)
- The user should be informed this has happened

**Review by stumpylog — CHANGES_REQUESTED** [review]

File: docker/compose/docker-compose.postgres.yml
Comment (Harald-Berghoff) [reply docker/compose/docker-compose.postgres.yml:1]: Fine for me, but didn't you write in #4033 that the line is expected in the postgres configuration? 

stumpylog
Aug 21, 2023
Maintainer

That's all loopback addresses, which I think means it would only trust connections to itself from itself. Something in their init must have gotten mixed up, as the default would be `host all all all trust". basically, it's missing addresses that are not local ones

If that line is expected, the option has to be added. 

But be clear: it isn't my specific setup. It is what is to be expected according to docker networking documentation and postgres image documentation.

File: install-paperless-ngx.sh (hunk 1 above)
Comment (Harald-Berghoff) [reply install-paperless-ngx.sh:82]: Any preferences on how this information should look like?

i.e. echo "Unable to detect timezone, defaulting to Etc/UTC"?

Comment (stumpylog) [reply install-paperless-ngx.sh:82]: That's fine

Comment (Harald-Berghoff) [reply install-paperless-ngx.sh]: solved with https://github.com/paperless-ngx/paperless-ngx/pull/6336/commits/41a5010bfef127f4976d0ebca6ffc1d34f695e6b

**Review by stumpylog — APPROVED** [review]
